### PR TITLE
[Voice to Content] Add eligibility checks

### PR DESCRIPTION
**Pull request by kean**
Fixes https://github.com/Automattic/wordpress-mobile/issues/103

## Architecture

I decided to continue keeping everything in the same View/ViewModel. It's small enough and it's easier to reuse stuff this way and make changes.

## Testing

**Happy Path**

- Tap "Post from Audio"
- ✅ Verify that is shows a loading state and the button "Record" is disabled
- ✅ Verify that if fetches eligibility info and shows the number of remaining requests (for unlimited plan, shows "Unlimited") and enables the "Record" button

**Not Eligible / Not Enough Requests**

- **Precondition**: Mock this link `if info.hasFeature, !info.isOverLimit {` or mock the network request to show the "not eligible scenario"
- Tap "Post from Audio"
- ✅ Verify that is shows an "upgrade required" view

## Known Issues

The `buttonUpgradeTapped` doesn't work as the upgrade URL is not correctly formatted.

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift [hunk 1]
```diff
@@ -1,36 +1,73 @@
...
+                    if case .welcome = viewModel.step, case .loading = viewModel.loadingState {
+                        ProgressView()
+                            .tint(.secondary)
+                            .controlSize(.small)
+                            .padding(.leading, 2)
+                    }
+                }
+            }
+            Spacer()
+            Button(action: viewModel.buttonCancelTapped) {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift]: Tapping on the button doesn't dismiss the view

**Review by momo-ozawa — COMMENTED** [review]
The close button doesn't dismiss the view, but otherwise works as described!

File: WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift]: Fixed

**Review by momo-ozawa — APPROVED** [review]

**Pull request merged**